src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerCommitImageFunctionalTest.groovy

```diff
@@ -0,0 +1,113 @@
+package com.bmuschko.gradle.docker.tasks.image
+
+import com.bmuschko.gradle.docker.AbstractGroovyDslFunctionalTest
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.UnexpectedBuildFailure
+
+class DockerCommitImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
+
+    def "can commit image"() {
+        given:
+        String commitImage = """
+            task commitImage(type: DockerCommitImage) {
+                dependsOn startContainer
+                targetContainerId createContainer.getContainerId()
+                author = "john doe"
+                message = "My image created"
+                tag = "myimage:latest"
+            }
+        """
+
+        buildFile << containerStart(commitImage)
+
+        when:
+        BuildResult result = build('commitImage')
+
+        then:
+        result.output.contains("Commiting image for container")
+    }
+
+    def "cannot commit image without container"() {
```

Comment:
I'd say this test case is not necessary as we are basically testing that the Gradle `Provider` API works as expected. You can remove this test case.

Reply:
The intention was to verify that the `DockerExistingContainer#containerId` is properly annotated, and that the Gradle input validation kicks in.

Happy to remove though.